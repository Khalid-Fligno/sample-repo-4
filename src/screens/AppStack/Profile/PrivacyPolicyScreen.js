import React from 'react';
import { StyleSheet, SafeAreaView, View, Text, ScrollView, Dimensions, Linking, Alert } from 'react-native';
import { db } from '../../../../config/firebase';
import Loader from '../../../components/Shared/Loader';
import colors from '../../../styles/colors';
import fonts from '../../../styles/fonts';

const { width } = Dimensions.get('window');

export default class PrivacyPolicyScreen extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      text: undefined,
      loading: false,
    };
  }
  componentDidMount = async () => {
    this.fetchText();
  }
  fetchText = () => {
    this.setState({ loading: true });
    db.collection('legalDocuments').doc('v3h3itZL99UHIpiILeVz')
      .get()
      .then(async (doc) => {
        if (doc.exists) {
          this.setState({ text: await doc.data().text, loading: false });
        } else {
          this.setState({ loading: false });
        }
      })
      .catch(() => {
        Alert.alert('Connection error', 'Please try again later');
        this.setState({ loading: false });
      });
  }
  openLink = (url) => {
    Linking.openURL(url);
  }
  render() {
    const { text, loading } = this.state;
    const sortedText = text && text.sort((a, b) => {
      return a.id - b.id;
    });
    const textDisplay = sortedText && sortedText.map((paragraph) => (
      <Text
        key={paragraph.id}
        onPress={() => paragraph.url && this.openLink(paragraph.url)}
        style={[
          styles.paragraph,
          paragraph.heading && styles.paragraphHeading,
          paragraph.url && styles.link,
        ]}
      >
        {paragraph.value}
      </Text>
    ));
    return (
      <SafeAreaView style={styles.safeContainer}>
        <View style={styles.container}>
          <ScrollView contentContainerStyle={styles.scrollView}>
            <Text style={styles.header}>
              Privacy Policy
            </Text>
            {textDisplay}
          </ScrollView>
          <Loader
            loading={loading}
            color={colors.charcoal.standard}
          />
        </View>
      </SafeAreaView>
    );
  }
}

const styles = StyleSheet.create({
  safeContainer: {
    flex: 1,
    backgroundColor: colors.black,
  },
  container: {
    flex: 1,
    backgroundColor: colors.white,
    alignItems: 'center',
  },
  scrollView: {
    width,
    padding: 15,
  },
  header: {
    fontFamily: fonts.bold,
    fontSize: 24,
    color: colors.charcoal.standard,
    marginBottom: 10,
  },
  paragraphHeading: {
    fontFamily: fonts.bold,
    fontSize: 14,
    color: colors.charcoal.dark,
    marginTop: 5,
    marginBottom: 8,
  },
  paragraph: {
    fontFamily: fonts.standard,
    fontSize: 14,
    color: colors.charcoal.standard,
    marginBottom: 8,
  },
  link: {
    fontFamily: fonts.standard,
    fontSize: 14,
    color: colors.blue.vivid,
    marginBottom: 8,
    textDecorationStyle: 'solid',
    textDecorationColor: colors.blue.vivid,
    textDecorationLine: 'underline',
  },
});
