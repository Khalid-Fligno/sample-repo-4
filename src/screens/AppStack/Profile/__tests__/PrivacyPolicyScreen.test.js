import React from 'react';
import renderer from 'react-test-renderer';
import PrivacyPolicyScreen from '../PrivacyPolicyScreen';

jest.useFakeTimers();

describe('PrivacyPolicyScreen', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<PrivacyPolicyScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
