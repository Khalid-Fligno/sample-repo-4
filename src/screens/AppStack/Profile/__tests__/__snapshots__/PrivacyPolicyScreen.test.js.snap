// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivacyPolicyScreen renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "rgb(0, 0, 0)",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(255, 255, 255)",
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        Object {
          "padding": 15,
          "width": 750,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "color": "rgb(38, 38, 40)",
              "fontFamily": "GothamBold",
              "fontSize": 24,
              "marginBottom": 10,
            }
          }
        >
          Privacy Policy
        </Text>
      </View>
    </RCTScrollView>
    <View
      style={
        Object {
          "backgroundColor": "rgb(255, 255, 255)",
          "flex": 1,
          "position": "absolute",
        }
      }
    >
      <Modal
        animationType="fade"
        hardwareAccelerated={false}
        onRequestClose={[Function]}
        supportedOrientations={
          Array [
            "landscape",
            "portrait",
          ]
        }
        transparent={true}
        visible={true}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "bottom": 0,
                "flex": 1,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.6)",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "height": 50,
                }
              }
            >
              <View
                animating={true}
                animationDuration={1200}
                animationEasing={[Function]}
                color="rgb(38, 38, 40)"
                hideAnimationDuration={200}
                interaction={true}
                renderComponent={[Function]}
                size={10}
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "rgb(38, 38, 40)",
                      "borderRadius": 5,
                      "height": 10,
                      "margin": 5,
                      "transform": Array [
                        Object {
                          "scale": 1,
                        },
                      ],
                      "width": 10,
                    }
                  }
                />
                <View
                  style={
                    Object {
                      "backgroundColor": "rgb(38, 38, 40)",
                      "borderRadius": 5,
                      "height": 10,
                      "margin": 5,
                      "transform": Array [
                        Object {
                          "scale": 1,
                        },
                      ],
                      "width": 10,
                    }
                  }
                />
                <View
                  style={
                    Object {
                      "backgroundColor": "rgb(38, 38, 40)",
                      "borderRadius": 5,
                      "height": 10,
                      "margin": 5,
                      "transform": Array [
                        Object {
                          "scale": 1,
                        },
                      ],
                      "width": 10,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </Modal>
    </View>
  </View>
</RCTSafeAreaView>
`;
